spec/gurney/cli_spec.rb

```diff
@@ -3,119 +3,156 @@
+            expect { Gurney::CLI.run('--client-hook'.split(' ')) }.not_to raise_error
+          end
+        end
+
+        it 'does not crash when receiving tags (BUGFIX)' do
+          expect_any_instance_of(Gurney::Api).not_to receive(:post_json)
+          run_with_branch_info '123abcdefaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa 123abcdefaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa refs/tags/v1'
+        end
+
+        context 'with main branch named "main"' do
+          let(:main_branch) { 'main' }
+
+          it 'works' do
+            expect_report(to: 'http://example.com/project/1/branch/main')
+            run_with_branch_info '123abcdefaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa 123abcaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa refs/heads/main'
+          end
         end
       end
+    end
 
+    context 'when the dependency files are nested in a subdirectory' do
+      include_context 'within a temporary git repository', path: 'spec/fixtures'
+
+      it 'reports all dependencies to the API' do
+        expect_any_instance_of(Gurney::Api).to receive(:post_dependencies)
+          .with(
+            dependencies: contain_exactly(
+              Gurney::Dependency.new(ecosystem: 'rubygems', name: 'byebug', version: '11.0.1'),
+              Gurney::Dependency.new(ecosystem: 'rubygems', name: 'httparty', version: '0.17.1'),
+              Gurney::Dependency.new(ecosystem: 'rubygems', name: 'mime-types', version: '3.3'),
+              Gurney::Dependency.new(ecosystem: 'rubygems', name: 'mime-types-data', version: '3.2019.1009'),
+              Gurney::Dependency.new(ecosystem: 'rubygems', name: 'multi_xml', version: '0.6.0'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: 'abbrev', version: '1.1.1'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: 'accepts', version: '1.3.4'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: 'accepts', version: '1.3.5'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: 'chalk', version: '5.4.1'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: 'commander', version: '13.1.0'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: 'lodash', version: '4.17.21'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: '@fortawesome/fontawesome-free', version: '5.15.4'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: 'bootstrap', version: '4.6.2'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: 'jquery', version: '3.7.1'),
+              Gurney::Dependency.new(ecosystem: 'npm', name: 'popper.js', version: '1.16.1'),
+              Gurney::Dependency.new(ecosystem: 'ruby', name: 'ruby', version: '2.3.8'),
+            ),
+            branch: 'master',
+            project_id: '1',
+            repo_path: instance_of(String),
+          ).and_return(double)
+
+        # We had to set the path to the configuration file per run parameters,
+        # as we didn't want to duplicate the fixtures
```

Comment:
Could we drop this comment? Calling Gurney like this is totally what I'd expect.